lib/Core/ExternalDispatcher.cpp

```diff
@@ -47,6 +50,13 @@ void *ExternalDispatcher::resolveSymbol(const std::string &name) {
         addr = llvm::sys::DynamicLibrary::SearchForAddressOfSymbol(str);
     }
 
+    if(!addr) {
+        addr = dlsym(RTLD_DEFAULT, str);
+        if(addr) {
+            llvm::sys::DynamicLibrary::AddSymbol(str, addr);
```

Comment:
This should work for symbol resolution. You may also need to extend function invocation to handle functions that take or return floating point data (float/double/long double). These may be passed in FP registers and the current implementation does not support that.

Reply:
I think I need to implement several function prototypes like "external_func_t" to deal with functions returning/taking floating point data. I guess I can get function return/parameter types from llvm::Function class. 
Is it a right approach to this?

And... is the original KLEE supporting external functions with floating point data? They generate stubs before calling external functions. Probably, the stub enables KLEE to support external functions with floating point data.

Thank you.

Reply:
Yeah, they generated stubs, but that's very expensive to do, it was showing on the profile and I removed them.
The KLEE version in S2E is very specialized, it does not need to handle all types of functions and can only restrict itself to those used by helpers. Therefore, you can hard-code all these external functions, e.g,: 
switch (functionName) {
  case "log": return logl(..);
  case "sin": ...
}

See the issue description to get a complete list of these functions.
It's important to have a whitelist though, that helps catching ones we don't support and that may have a weird calling convention.